src/screens/TopicPageScreen/elements/BottomSheetFollow.tsx

```diff
@@ -24,25 +22,25 @@ interface BottomSheetFollowProps {
   memberCount: number;
   setMemberCount: (memberCount: number) => void;
   onClose: () => void;
-  getTopicDetail: (topicName: string) => void;
   followType: Follow;
   setFollowType: (followType: Follow) => void;
+  topicId: string;
 }
 
 const BottomSheetFollow = forwardRef((props: BottomSheetFollowProps, ref: Ref<RBSheet>) => {
   const {
+    topicId,
     topicName,
     memberCount,
     setMemberCount,
     isFollow,
     setIsFollow,
     followType,
     setFollowType,
-    getTopicDetail,
     onClose
   } = props;
-  const [, discoveryDispatch] = (React.useContext(Context) as any)?.discovery;
   const {followTopic} = useChatClientHook();
+  const {updateFollowTopicDiscoveryContext} = useDiscovery();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The removal of `getTopicDetail` from the props and its usage might cause issues if it was used to fetch and update the topic details in the parent component. Ensure that this functionality is covered elsewhere or not needed anymore.

```diff
- getTopicDetail: (topicName: string) => void;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->